doc/fiware-tour-guide-application-a-tutorial-on-how-to-integrate-the-main-fiware-ges/publishing-historical-data.md

```diff
@@ -1 +1,206 @@
+                                   [-p <password> | --mysql-password <password>]
+
+Apply configuration changes for cygnus.
+
+Command options:
+
+  -h  --help                         Show this help.
+  -u  --mysql-user <username>        Set the MySQL database user to use.
+                                     Default value is 'root'.
+  -p  --mysql-password <password>    Set the MySQL database password to use.
+                                     Default value is 'mysql'.
+```
+
+i.e. to change the MySQL credentials used for Cygnus to `myuser` and `mypassword`, issue the following command:
+```
+$ ./tour-guide configure cygnus --mysql-user myuser --mysql-password mypassword
+```
+
+
+This will update the `docker-compose.yml` file and set the following variables for the Cygnus container:
+
+```
+- CYGNUS_MYSQL_USER=myuser
+- CYGNUS_MYSQL_PASS=mypassword
+```
+
+This variables will be used by the Cygnus container to update the following MySQL credentials on the `agent.conf` configuration file when the container starts:
+```
+cygnus-ngsi.sinks.mysql-sink.mysql_username = myuser
+cygnus-ngsi.sinks.mysql-sink.mysql_password = mypassword
+```
+For more information on other available configuration options for the Cygnus container, see [Using the image](https://github.com/telefonicaid/fiware-cygnus/blob/master/docker/cygnus-ngsi/README.md#section3) on the Cygnus repository.  For more complex configuration options, see the [official documentation for Cygnus](https://fiware-cygnus.readthedocs.io/en/latest/).
+
+Besides the Cygnus container modifications, the following variables will be set for the MySQL container:
+
+```
+- MYSQL_USER=myuser
+- MYSQL_PASSWORD=mypassword
+- MYSQL_DATABASE=tourguide
+```
+
+The `MYSQL_DATABASE` variable defines the database to use.  As we are using a user with no superuser privileges, we need to tell the MySQL container to create the database for us.  The database name will be the same as the 'Fiware Service' used for the sensors, which defaults to `tourguide` for the TourGuide-App.  If we were using the `root` user, there'd be no need to specify the database name, as that would be created on demand.
+
+
+## Subscriptions
+
+After the system is configured and running, we need to tell Orion that we want it to send the data as it changes. To do this we need to register a subscription on Orion for Cygnus.
+
+Imagine we want to store the temperature changes on a restaurant over a period of time.  We then need to tell Orion to send us the information whenever any of the temperature sensors value for the restaurant changes. To do this we need the following information:
+
+- Id of the restaurant, e.g. `0115206c51f60b48b77e4c937835795c33bb953f`
+- the type of the sensor, e.g. `temperature`
+- the room of the sensor, e.g. `kitchen`
+
+With this information, we will send a POST request to `http://${ORION_HOST}:${ORION_PORT}/v1/subscribeContext` with a payload like this:
```

Comment:
Yes. There is no precondition in this case, because we will do the request against `http://localhost:1026` and inside Orion container, the Cygnus host is known.